### PR TITLE
ft(view-cart) buyer should be able to view items in the cart

### PR DESCRIPTION
**Pull request by chrill-christella**
### what does this PR do?

 implement the GET endpoint for the buyer to view items in cart

### How should it be manually tested

• clone the repo using git clone repository's URL
• change branch git checkout ft-view-cart
• run npm install, npm run migrate:up and npm run db:seeds
• run the endpoint within the swagger or send get request http://localhost:<PORT>/api/cart/getAll

### What are the relevant pivotal tracker stories?

https://trello.com/c/99dTN28l

### CODE REVIEW

File: src/controllers/cartController.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    }
+  };
+
+  static async getCartItems(req, res) {
+    try {
+      const Cart = await cart.findOne({ where: { buyerId: req.user.id } });
+      if (!Cart) {
+        res.status(401).json({ status: "fail", message: "Cart is Empty" });
+      } else {
+        res.status(304).json({ status:"success", Cart });
```
Comment (wakaflorien) [inline src/controllers/cartController.js]: - return correct json response

```suggestion
        res.status(200).json({ status:"success", message: "cart items", data:Cart });
```

File: test/buyer.test.js [hunk 2]
```diff
@@ -97,4 +96,43 @@ describe("testing buyer signup",() =>{
...
+  })
+})
+
+  describe('/getCartItems endpoint', () => {
+    test('should return array of all items', async () => {
+        const res = await request.get('/api/cart/getAll').set('token',  `Bearer ${token}`);
+        expect(res.statusCode).toBe(304);
+       
+    });
+  })
```
Comment (wakaflorien) [inline test/buyer.test.js]: - write more tests for different scenarios that may happen when performing this action.

**Review by wakaflorien — CHANGES_REQUESTED** [review]

**Review by Shumba250 — APPROVED** [review]

**Review by sostene12 — CHANGES_REQUESTED** [review]
- get the user cart is showing empty
-pushed latest changes, you can check again

File: src/controllers/cartController.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+      }
+     
+      catch (error) {
+        return res.status(500).json({ status: "error", error: error.message });
+    }
+  };
+
+  static async getCartItems(req, res) {
+    try {
+      const Cart = await cart.findOne({ where: { buyerId: req.user.id } });
```
Comment (wakaflorien) [inline src/controllers/cartController.js]: - this will return all available carts for the specific user. what if we want single cart with it's items?

**Review by wakaflorien — APPROVED** [review]

**Review by sostene12 — APPROVED** [review]

**Review by wakaflorien — APPROVED** [review]

**Pull request merged**